src/main/java/com/aws/greengrass/deployment/exceptions/NonRetryableDeploymentTaskFailureException.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment.exceptions;
+
+public class NonRetryableDeploymentTaskFailureException extends DeploymentException {
```

Comment:
Could you explain why you're bringing this back? There are no retries at deployment level anymore, individual steps within a deployment workflow has it's own retry policy like download/resolution etc

Reply:
Each step (like resolution/download) throws a different type of exception and the catch block in DefaultDeploymentTask was getting large. Created this to be parent exception for all exceptions that are thrown inside DefaultDeploymentTask.

Reply:
If you just want a parent exception, don't call it NonRetryable-, since this is kind of misleading given that there are no retries based on the exception type like there used to be

Reply:
Synced offline, decided to use DeploymentTaskFailureException.